src/Wallabag/CommentBundle/Controller/WallabagCommentController.php

```diff
@@ -0,0 +1,168 @@
+    {
+        $data = json_decode($request->getContent(), true);
+
+        $this->validateAuthentication();
+
+        $em = $this->getDoctrine()->getManager();
+
+        $comment = new Comment($this->getUser());
+
+        $comment->setText($data['text']);
+        if (array_key_exists('quote', $data)) {
+            $comment->setQuote($data['quote']);
+        }
+        if (array_key_exists('ranges', $data)) {
+            $comment->setRanges($data['ranges']);
+        }
+        $comment->setUpdated(new \DateTime());
+
+        $comment->setEntry($entry);
+
+        $em->persist($comment);
+        $em->flush();
+
+        $json = $this->get('serializer')->serialize($comment, 'json');
+
+        return $this->renderJsonResponse($json);
+    }
+
+    /**
+     * Updates a comment.
+     *
+     * @ApiDoc(
+     *      requirements={
+     *          {"name"="comment", "dataType"="string", "requirement"="\w+", "description"="The comment ID"}
+     *      }
+     * )
+     *
+     * @return Response
+     */
+    public function putAnnotationAction($idcomment, Request $request)
+    {
+        $this->validateAuthentication();
+
+        $data = json_decode($request->getContent(), true);
+
+        $comment = $this
+                ->getDoctrine()
+                ->getRepository('WallabagCommentBundle:Comment')
+                ->findCommentById($idcomment);
+
+        if (!is_null($data['text'])) {
+            $comment->setText($data['text']);
+            $comment->setUpdated(new \DateTime());
+        }
+
+        $em = $this->getDoctrine()->getManager();
+        $em->flush();
+
+        $json = $this->get('serializer')->serialize($comment, 'json');
+
+        return $this->renderJsonResponse($json);
+    }
+
+    /**
+     * Removes a comment.
+     *
+     * @ApiDoc(
+     *      requirements={
+     *          {"name"="comment", "dataType"="string", "requirement"="\w+", "description"="The comment ID"}
+     *      }
+     * )
+     *
+     * @return Response
+     */
+    public function deleteAnnotationAction($idcomment)
+    {
+        $this->validateAuthentication();
+
+        $comment = $this
+                ->getDoctrine()
+                ->getRepository('WallabagCommentBundle:Comment')
+                ->findCommentById($idcomment);
+
+        $em = $this->getDoctrine()->getManager();
+        $em->remove($comment);
+        $em->flush();
+
+        $json = $this->get('serializer')->serialize($comment, 'json');
+
+        return $this->renderJsonResponse($json);
+    }
+
+    /**
+     * Send a JSON Response.
+     * We don't use the Symfony JsonRespone, because it takes an array as parameter instead of a JSON string.
+     *
+     * @param string $json
+     *
+     * @return Response
+     */
+    private function renderJsonResponse($json, $code = 200)
+    {
+        return new Response($json, $code, array('application/json'));
+    }
+
+    private function validateAuthentication()
```

Comment:
You should better add a new rule in the `security.yml` file related to your url instead.
https://github.com/wallabag/wallabag/blob/v2/app/config/security.yml#L54